src/candidates/profile/cv_upload/CvUpload.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div class="cv-upload">
+    <div id="cv-upload-content">
+      <file-upload
+        id="cv-upload-form"
+        ref="upload"
+        :drop="true"
+        :drop-directory="false"
+        :multiple="false"
+        @input-file="inputFile"
+        @input-filter="inputFilter">
+        <p>Please drop your PDF file here or click to upload one</p>
+      </file-upload>
+    </div>
+  </div>
+</template>
+
+<script>
+// NOTE:
+//
+// The VueUploadComponent documentation can be found here:
+//
+// https://lian-yue.github.io/vue-upload-component/#/en/documents
+//
+// --- 2018-03-12 - lilsweetcaligula
+//
+import VueUploadComponent from 'vue-upload-component';
+
+import _ from 'lodash';
+import path from 'path';
+
+// NOTE:
+//
+// Importing 'pdfjs-dist/webpack' will configure the worker for us.
+//
+// If you want to configure one yourself, then import 'pdfjs-dist',
+// copy the worker from  ./node_modules/pdfjs-dist/build/pdf.worker.js
+// to the 'static' folder and set pdfJs.GlobalWorkerOptions.workerSrc
+// like this:
+//
+// ```
+// pdfJs.GlobalWorkerOptions.workerSrc = './../../../../static/pdf.worker.js';
+// ```
+//
+// --- 2018-03-12 - lilsweetcaligula
+//
+import pdfJs from 'pdfjs-dist/webpack';
+
+const readPdfPagesContent = async (...args) => {
+  const pdf = await pdfJs.getDocument(...args);
+
+  const pagePromises = _.range(1, pdf.numPages + 1).map(pageNum => {
+    return pdf.getPage(pageNum);
+  });
+
+  const pages = await Promise.all(pagePromises);
+
+  const contentPromises = pages.map(page => {
+    return page.getTextContent({ normalizeWhitespace: true });
+  });
+
+  const contents = await Promise.all(contentPromises);
+
+  const pagesContent = contents.map(content => {
+    const pageText = content.items
+      .reduce((acc, item) => acc + item.str, '');
```

Comment:
Concatenating strings like that can actually be a performance bottleneck. So if the upload ever needs to be optimized, this may be a good place to start.

Reply:
Can you add this as `NOTE` comment?

Reply:
1de605f18781bb9fd8340c0d0cbac8b75a78bc2f